zhaquirks/thirdreality/night_light.py

```diff
@@ -0,0 +1,157 @@
+"""Third Reality night light zigbee devices."""
+import logging
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.security import IasZone
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Identify,
+    LevelControl,
+    MultistateInput,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+from zigpy.zcl.clusters.lighting import Color
+from zigpy.zcl.clusters.lightlink import LightLink
+from zigpy.zcl.clusters.manufacturer_specific import ManufacturerSpecificCluster
+from zigpy.zcl.clusters.measurement import (
+    OccupancySensing,
+    IlluminanceMeasurement,
+)
+
+from zhaquirks import (
+    Bus,
+    LocalDataCluster,
+    MotionOnEvent,
+    OccupancyWithReset,
+    QuickInitDevice,
+)
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+    ATTRIBUTE_ID,
+    ATTRIBUTE_NAME,
+    COMMAND_ATTRIBUTE_UPDATED,
+    COMMAND_TRIPLE,
+    UNKNOWN,
+    VALUE,
+    ZHA_SEND_EVENT,
+    ZONE_STATUS,
+    ZHA_SEND_EVENT,
+    ZONE_STATUS_CHANGE_COMMAND,
+)
+from zhaquirks.thirdreality import THIRD_REALITY
+ThirdReality_Specific_CLUSTER_ID = 0xFC00
+MOTION_TYPE = 0x0002
+ZONE_TYPE = 0x0001
+MOTION_REPORTED = "motion_reported"
+
+class OccupancyCluster(OccupancyWithReset):
+    """Occupancy cluster."""
+
+class MotionCluster(LocalDataCluster, MotionOnEvent):
+    """Motion cluster."""
+
+    _CONSTANT_ATTRIBUTES = {ZONE_TYPE: MOTION_TYPE}
+    reset_s: int = 30
+
+
+_LOGGER = logging.getLogger(__name__)
+
+class ThirdRealitySpecificCluster(CustomCluster):
+    """privatecluster, only used to relay ias zone information to ias zone Cluster."""
+
+    cluster_id = ThirdReality_Specific_CLUSTER_ID
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if value is not None and value >= 0:
+            self.endpoint.device.motion_bus.listener_event("ias_zone_reported", value)
```

Comment:
1. Do you know what attribute ID is used for the motion event? (so `attrid`, does it differ for on/off?)
2. Do you know what the possible values are? (so `value` for on/off)

Also, we probably don't need the `motion_bus` anymore and can just update the attribute on that other cluster from here:
```python
self.endpoint.ias_zone.update_attribute(IasZone.AttributeDefs.zone_status.id, value)
```

Reply:
Thanks. The device used private cluter(0xfc00) to replace ias zone cluster(0x0500).
In addition, the device used like ias zone  ZoneStatus attribute(0x0002) and the value is same as ias zone, ZoneStatus（bitmap16）. 
By the way, I wrote motion_bus by imitating the examples of Xiaomi on the website(https://github.com/zigpy/zha-device-handlers).

Reply:
> In addition, the device used like ias zone ZoneStatus attribute(0x0002) and the value is same as ias zone,

So just to confirm, attribute id `0x0002` is used on the private/manufacturer specific cluster?

Right now, the code forwards all "attribute updates" on that cluster to `zone_status` (`0x0002`) on the `IasZone` cluster.
It would be better to specifically check that only the right attribute (id) is forwarded to the `IasZone` cluster.
That way, there won't be any issues in the future (if, for example, future firmware causes other attribute updates on that cluster that aren't motion detection).

> By the way, I wrote motion_bus by imitating the examples of Xiaomi on the website

Yeah, that example is a bit outdated. Sorry about that. It really needs to be updated.